pkg/standalone-utils/contracts/PoolSwapFeeHelper.sol

```diff
@@ -5,102 +5,38 @@ pragma solidity ^0.8.24;
 import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 
 import { IPoolSwapFeeHelper } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IPoolSwapFeeHelper.sol";
-import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
 import { IVaultErrors } from "@balancer-labs/v3-interfaces/contracts/vault/IVaultErrors.sol";
 import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
 
-contract PoolSwapFeeHelper is IPoolSwapFeeHelper, SingletonAuthentication {
-    using EnumerableSet for EnumerableSet.AddressSet;
+import { PoolHelperCommon } from "./PoolHelperCommon.sol";
 
-    EnumerableSet.AddressSet private _pools;
+contract PoolSwapFeeHelper is IPoolSwapFeeHelper, PoolHelperCommon {
+    using EnumerableSet for EnumerableSet.AddressSet;
 
-    constructor(IVault vault) SingletonAuthentication(vault) {
+    constructor(IVault vault) PoolHelperCommon(vault) {
         // solhint-disable-previous-line no-empty-blocks
     }
 
     /***************************************************************************
                                     Manage Pools
     ***************************************************************************/
 
-    /// @inheritdoc IPoolSwapFeeHelper
-    function addPools(address[] calldata newPools) external authenticate {
-        uint256 length = newPools.length;
-        IVault vault = getVault();
-
-        for (uint256 i = 0; i < length; i++) {
-            address pool = newPools[i];
-
-            // Ensure the address is a valid pool.
-            if (vault.isPoolRegistered(pool) == false) {
-                revert IVaultErrors.PoolNotRegistered(pool);
-            }
-
-            // Pools cannot have a swap fee manager.
-            if (vault.getPoolRoleAccounts(pool).swapFeeManager != address(0)) {
-                revert PoolHasSwapManager(pool);
-            }
-
-            if (_pools.add(pool) == false) {
-                revert PoolAlreadyInSwapFeeSet(pool);
-            }
-
-            emit PoolAddedToSwapFeeSet(pool);
-        }
-    }
-
-    /// @inheritdoc IPoolSwapFeeHelper
-    function removePools(address[] memory pools) public authenticate {
-        uint256 length = pools.length;
-        for (uint256 i = 0; i < length; i++) {
-            address pool = pools[i];
-            _ensureKnownPool(pool);
-            _pools.remove(pool);
-
-            emit PoolRemovedFromSwapFeeSet(pool);
-        }
-    }
-
     /// @inheritdoc IPoolSwapFeeHelper
     function setStaticSwapFeePercentage(address pool, uint256 swapFeePercentage) public authenticate {
```

Comment:
(for another PR): should we have a setter for multiple pools / multiple fee percentages as well? (parallel arrays)

Reply:
Sounds like a question to ask the people who want to use this. Would it be more common to set a bunch of pools to the same fee, so only the first argument would be an array? Is there a use case for setting a bunch of pools to random different values (maybe some kind of Gauntlet-like fee setter)? If those pools are all from the same factory, you would probably use the factory-based fee setter.

Open to all opens, but we should probably verify there's a use case before we just add stuff.

Reply:
the use case would be pools that are managed by a third party via the helper, but somewhat manually (as opposed to ezkl where we're setting the fee pool by pool).

Reply:
Should we have a setter with an array of pools and a single value, and also a setter with parallel arrays?